core/changestream/eventsource.go

```diff
@@ -48,33 +47,11 @@ func NewTxnRunnerFactory(f WatchableDBFactory) database.TxnRunnerFactory {
 // an error.
 type WatchableDBFactory = func() (WatchableDB, error)
 
-// WatchableModelDBFactory provides an interface for getting a
-// database.TxnRunner or error that is scoped to a specific model.
-type WatchableModelDBFactory struct {
-	WatchableDBFactory
-	ModelUUID model.UUID
-}
-
 // NewWatchableDBFactoryForNamespace returns a WatchableDBFactory
 // for the input namespaced factory function and namespace.
-func NewWatchableDBFactoryForNamespace[T WatchableDB](
-	ns string,
-	f func(string) (T, error),
-) WatchableDBFactory {
+func NewWatchableDBFactoryForNamespace[T WatchableDB](f func(string) (T, error), ns string) WatchableDBFactory {
```

Comment:
Since isolating service construction to the service factory worker, this is only called there. Worth relocating?

Reply:
Leave it for now, but I might sweep back around.